src/app/domain/auth/AuthServices.ts

```diff
@@ -0,0 +1,52 @@
+import bcrypt from "bcrypt";
+
+import { BadRequestError, InternalServerError } from "routing-controllers";
+
+import { User } from "models/user";
+import { IAuth } from "types/auth";
+import { createToken } from "utils/token";
+import { CustomError } from "utils/customError";
+
+const SALT = 10;
+
+export default class AuthServices {
+  async signIn({ email, password }: IAuth) {
+    try {
+      const userData = await User.findOne({ email });
+
+      if (!userData) throw new BadRequestError("Incorrect email or password!");
```

Comment:
it's better to use User not found to clarify that the user needs to change the email